config/internal/apiserver/deployment.yaml.tmpl

```diff
@@ -201,7 +201,7 @@ spec:
             - mountPath: /etc/tls/private
               name: proxy-tls
         {{ end }}
-      serviceAccountName: ds-pipeline-{{.Name}}
+      serviceAccountName: {{.APIServerDeploymentName}}
```

Comment:
this is tied to: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/internal/apiserver/sa_ds-pipeline.yaml.tmpl#L4 

so if we change the service account name here, we need to do the same for the SA metadata.name
at which point, it is probably not appropriate to call it `APIServerDeploymentName` maybe something like `APIServerDefaultResourceName`. Same for persistent agent/swf.